### DISCUSSION TITLE
Getting a "Fixed" option to dismiss codeql alerts.

### DISCUSSION THREAD

**Discussion by mmishra-ias** [community/community#77075] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

👋 all. At situations where CodeQL is unable to understand if a vulnerability is fixed, can we have a "Fixed" option to dismiss alert or is there another way that can be followed.(I feel a "Fixed" option should have been a better go to.)

Thanks and regards,
Mayukh

**Comment by jketema** [comment]
Hi @mmishra-ias,

The page with the details for an alert has some dismiss alert options in the upper right corner. You can close the alert with one of the options there. The "False positive" option seems most appropriate from what you are are describing.

![Screenshot 2023-11-27 at 09 52 03](https://github.com/community/community/assets/93738568/cc50c2be-c9a1-4287-821a-e18194d9833b)

If you have a nice example of a case where you believe an error should not be reported, I'd also encourage you to open a false positive issue in the [CodeQL repo](https://github.com/github/codeql).

**Reply by mmishra-ias** [reply]
Hi @jketema,

Thanks for the insight. I do know these options but when there are situations where CodeQL was unable to mark a vulnerability as fixed, even when it has been mitigated, a fixed option would be handy. We had an issue while moving up to CodeQL CLI version and had hundreds of vulnerabilities with same issue(Now fixed). For future ease, I just think there should be one option to mark a vulnerability as fixed.

**Reply by jketema** [reply]
I believe alerts should automatically close in the case you describe. If that's not the case that seems like a bug. Let me check internally.

By the way, if this is on a public repository, a link to the repository can help us with the diagnosis.

**Reply by jketema** [reply]
I've asked internally. It can happen that the alerts are not closed automatically when something changed in your workflow configuration. For example, you renamed the workflow, or you add/changed the category name. Note that it is possible to delete old configurations.

It's very hard to tell what is going on in your case though without access to the actual alerts/repository.
